osm2gtfs/core/osm_connector.py

```diff
@@ -514,21 +601,22 @@ def _generate_shape(self, route_variant, query_result_set):
     def _is_valid_stop_candidate(self, stop):
         """Helper function to check if a stop candidate has a valid tagging
 
-        Returns True or False
+        :return bool: Returns True or False
 
         """
+        valid = False
         if 'public_transport' in stop.tags:
             if stop.tags['public_transport'] == 'platform':
-                return True
+                valid = True
             elif stop.tags['public_transport'] == 'station':
-                return True
-        elif 'highway' in stop.tags:
+                valid = True
+        if 'highway' in stop.tags:
             if stop.tags['highway'] == 'bus_stop':
-                return True
-        elif 'amenity' in stop.tags:
+                valid = True
+        if 'amenity' in stop.tags:
             if stop.tags['amenity'] == 'bus_station':
```

Comment:
All of this function seems to be very specific to buses, we should include cases for other public transport services too, such as `railway=station`, `railway=tram_stop`, ... (I'm sure there are a lot more)

Reply:
Yes, I really see here a good point in what you are saying. But I think this should be the topic of a separate PR, as it is not directly connected to the scope of this here.

Reply:
Created #100.